src/lib/classes/saml_admin.php

```diff
@@ -183,16 +211,30 @@ public function get_saml_status()
         }
       }
     }
-    
-    if(file_exists(constant('SAMLAUTH_CONF') . '/certs/' . get_current_blog_id() . '/' . get_current_blog_id() . '.cer') && file_exists(constant('SAMLAUTH_CONF') . '/certs/' . get_current_blog_id() . '/' . get_current_blog_id() . '.key'))
+
+    /*
+     * Check if public and private keys are set in the database config
+     * otherwise check for cert files.
+     */
+    if(
```

Comment:
There's quite a bit of boolean logic going into this `if` statement and it's become too difficult to read. I think section could benefit from moving a few of the portions of this logic into separate named variables so the final argument to the `if` statement is a simple, non-nested boolean operation.